workitem.go

```diff
@@ -268,13 +184,9 @@ func (c *WorkitemController) Delete(ctx *app.DeleteWorkitemContext) error {
 // response resource object by jsonapi.org specifications
 func ConvertJSONAPIToWorkItem(appl application.Application, source app.WorkItem2, target *app.WorkItem) error {
 	// construct default values from input WI
-	var version = -1
-	if source.Attributes["version"] != nil {
-		v, err := strconv.Atoi(fmt.Sprintf("%v", source.Attributes["version"]))
-		if err != nil {
-			return errors.NewBadParameterError("data.attributes.version", source.Attributes["version"])
-		}
-		version = v
+	version, err := getVersion(source.Attributes["version"])
+	if err != nil {
+		return err
```

Comment:
Maybe `return errs.WithStack(err)` or `return errs.Wrap(err, "some more context information")` is more helpful? At least that should be the pattern in order to have better error traces.